From e67189e107c188c13f1dc6e81c941f7e8e820d4b Mon Sep 17 00:00:00 2001
From: Ralph Boehme <slow@samba.org>
Date: Mon, 27 Jun 2016 11:39:47 +0200
Subject: [PATCH 6/8] s3/smbd: make get_ea_dos_attribute() public

Needed in a subsequent commit to read the stored dosmode for a file.

Bug: https://bugzilla.samba.org/show_bug.cgi?id=11992

Signed-off-by: Ralph Boehme <slow@samba.org>
---
 source3/smbd/dosmode.c | 6 +++---
 source3/smbd/proto.h   | 3 +++
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/source3/smbd/dosmode.c b/source3/smbd/dosmode.c
index 1574166..42b869c 100644
--- a/source3/smbd/dosmode.c
+++ b/source3/smbd/dosmode.c
@@ -257,9 +257,9 @@ static uint32_t dos_mode_from_sbuf(connection_struct *conn,
  This can also pull the create time into the stat struct inside smb_fname.
 ****************************************************************************/
 
-static bool get_ea_dos_attribute(connection_struct *conn,
-				 struct smb_filename *smb_fname,
-				 uint32_t *pattr)
+bool get_ea_dos_attribute(connection_struct *conn,
+			  struct smb_filename *smb_fname,
+			  uint32_t *pattr)
 {
 	struct xattr_DOSATTRIB dosattrib;
 	enum ndr_err_code ndr_err;
diff --git a/source3/smbd/proto.h b/source3/smbd/proto.h
index 9b9c924..51661c0 100644
--- a/source3/smbd/proto.h
+++ b/source3/smbd/proto.h
@@ -259,6 +259,9 @@ int dos_attributes_to_stat_dos_flags(uint32_t dosmode);
 uint32_t dos_mode(connection_struct *conn, struct smb_filename *smb_fname);
 int file_set_dosmode(connection_struct *conn, struct smb_filename *smb_fname,
 		     uint32_t dosmode, const char *parent_dir, bool newfile);
+bool get_ea_dos_attribute(connection_struct *conn,
+			  struct smb_filename *smb_fname,
+			  uint32_t *pattr);
 NTSTATUS file_set_sparse(connection_struct *conn,
 			 struct files_struct *fsp,
 			 bool sparse);
-- 
2.9.0

